### DISCUSSION TITLE
Multiple sites in one org-file?

### DISCUSSION THREAD

**Discussion by patmaddox** [kaushalmodi/ox-hugo#628] [Q&A]
Is it possible to publish multiple hugo sites from a single org file?

This syntax exports everything under `website1`:

```
* website 1
#+hugo_base_dir: website1
** article 1
* website 2
#+hugo_base_dir: website2
** article 2
```

Ultimately I would like to be able to link from articles in `website1` to ones in `website2` and have the links prefixed with the domain name (e.g. `http://website1.com/article1`). But for the time being, being able to export them to separate dirs will be good.

Also I'd want some kind of wim export that exports all of the subtrees in a particular site, or all subtrees in the file.

**Comment by kaushalmodi** [comment]
>  it possible to publish multiple hugo sites from a single org file?

No, each site needs a separate file. 


> Ultimately I would like to be able to link from articles in website1 to ones in website2 and have the links prefixed with the domain name (e.g. http://website1.com/article1

That will need a non-trivial amount of work, and I don't plan on adding support for that. Right now, ox-hugo exports links using a simple but fairly robust way.. it exports cross-post links as Hugo relref shortcodes.

Reasons:
- ox-hugo then doesn't need to derive full paths to those posts; it only needs the post's basename. 
- The job of finding the full path is offloaded to Hugo using relref shortcode, but Hugo does actually knows the path resolution better!
- relref comes with auto link checking feature. If you rename a post's file name, the relref will break, and Hugo will fail to build using that error. Such issues would go uncaught for a longer time if relref is not used. 

> Also I'd want some kind of wim export that exports all of the subtrees in a particular site,

Ox-hugo doesn't provide a way to export all the files, but the nuances of exporting multiple files is different for different people. You can find one example of doing this here: https://github.com/kaushalmodi/ox-hugo/discussions/585#discussioncomment-2335203

> or all subtrees in the file.

This is already supported using the default `C-c C-e H A` binding. See https://ox-hugo.scripter.co/#export-bindings.

---

Summary:

1. Ox-hugo won't allow specifying multiple `#+hugo_base_dir` in the same file. It will add unnecessary complication for little gain. Please have a separate Org file for each site. 
2. Cross-linking across Hugo sites won't be supported by the link formatting of the exporter. I'd suggest asking for ideas on how you can maintain links between 2 Hugo sites on https://discourse.gohugo.io/.
